### DISCUSSION TITLE
How to move from Prisma 2.12 to latest version without a reset in production?

### DISCUSSION THREAD

**Discussion by geosigno** [prisma/prisma#12347] [Q&A]
Hello,

We are having a production projet that is running with Prisma 2.12, and now it's the time for us to move to latest Prisma v3 :) 

I tried every scenario I could see on the doc to manage this, but it always ask me at some point to reset the DB, which can be done on dev environnement but definitely not on production.

Important note, we are not having any migrations files, because we were deleting them gradually to avoid conflict between team members. I know this is a bad practice, and the v3 will be for us a motivation to respect the normal prisma workflow moving forward.

So I started with a `prisma db pull` to have the latest rules reflected in the `schema.prisma`. So I got dozens of changes in my `schema.prisma`, some examples below:

before
```
stripeId           String?             @unique
storageSize               Float?   @default(0)
id          String  @id @default(cuid())
```

after
```
 stripeId           String?             @unique(map: "Account.stripeId_unique")
storageSize               Decimal? @default(0)
id          String  @id @default(cuid()) @db.VarChar(30)
```
etc. etc.

Then how can i manage to update the DB without removing the data in it? Even if I reset the dev, I got a migration file that is re-creating all the tables, which fails in production because the tables already exist.

**Comment by nurul3101** [comment]
Hey @geosigno 👋 

Did you try [baselining your database](https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/baselining)?

This is the recommended approach for production databases that cannot be reset. Baselining solves this problem by telling Prisma Migrate to pretend that the initial migration(s) have already been applied.
You can create an initial migration file with the current database state and mark it as resolved as suggested in the guide.

**Reply by geosigno** [reply]
Thanks for the reply.

The baseline job has worked I think...

But there is something I am not clear with. My Dev database is a perfect replica of my prisma.schema (because it was reset) with the new syntax etc. But in my Prod database, I kinda lie to it saying that the initial migration has already been done, which it hasn't. So, if I understand correctly, this means that my Dev and Prod databases are not exactly the same, from an architectural point of view. Can't this lead to a bigger problem later?

**Comment by geosigno** [comment]
This is still not working, here what I did:

1. install prisma 3.14
2. `prisma migrate dev --name initial-migration --create-only`
3. Prisma said there is a Drift detected, and it asks me to delete the all DB? So the "--create-only" is not working. I am in production so what should I do if I don't want to delete? For the sake of testing I said yes to restore the all db.
4.  `npx prisma migrate resolve --applied 20220513134214_initial_migration`

Then for the sake of testing (again), i tried a `npx prisma dev --name test` and I should see nothing but it created a new migration that create all the table and ask me to reset again the DB.

How can this became so difficult while it was working like a charm with prisma 2.12?

**Reply by janpio** [reply]
What about the baselining step that is required in between as @nurul3101 mentioned? Did that not work for you?

Prisma Migrate changed a lot between 2.12 and now - it is a completely different tool that works totally different and supports so much more things that you really, really want to work.

**Reply by geosigno** [reply]
The baseline is not working because it asks to reset the DB, even with the  --create-only (there is already a [bug](https://github.com/prisma/prisma/issues/6579) open).

I actually managed to get around this bug by using prisma diff explained by @pantharshit00  [here](https://tagmerge.com/issue/prisma/prisma/11555).

**Reply by janpio** [reply]
That other issue is not about applying the baseline process. For that you create a SQL dump of your database, and tell Migrate that the database is at that state. But what you linked to for Harshit is baselining, so seems you ended up doing the right thing.

**Reply by geosigno** [reply]
I followed the doc about baseline a database, and the first step is « Ensure that you have [updated your development environment and generated initial migrations](https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/add-prisma-migrate-to-a-project#update-the-development-environment) » . So to generate the initial migration it said to run
`prisma migrate dev --name initial-migration --create-only’… So we are back to square one where is it not possible to create a migration without resetting the DB (at least in my case).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
